pkg/stub/config.go

```diff
@@ -303,8 +285,7 @@ func (h *Handler) ProcessManagementField(cfg *v1.Config) (bool, bool, error) {
 	case operatorsv1api.Managed:
 		if cfg.Spec.ManagementState != cfg.Status.ManagementState {
 			logrus.Println("management state set to managed")
-			if cfg.Spec.InstallType == v1.RHELSamplesDistribution &&
-				cfg.ConditionFalse(v1.ImportCredentialsExist) {
+			if cfg.ConditionFalse(v1.ImportCredentialsExist) {
 				h.copyDefaultClusterPullSecret(nil)
```

Comment:
Along the lines of my earlier comment about the CredsExist condition, seems like we should just always do this.  But the copy has to be smart enough not to update the openshift namespace secret if it's unchanged.

Reply:
Per above, I say we keep the condition, and then the copy can be simple.

Reply:
Perhaps we just *NEVER* do this here on the config watch path, based on your other response, and just wait for the secret relist/resync to copy

Reply:
> Perhaps we just _NEVER_ do this here on the config watch path, based on your other response, and just wait for the secret relist/resync to copy

This one is a bit different since you're going from unmanaged to managed, so it's possible the secret doesn't currently exist in the openshift namespace.  We don't want to wait 10 mins to hit a resync, so something needs to explicitly try to get us back to a working state (similar to bootstrap).

but we can definitely tackle this in a separate PR.  I just think we want to revisit the whole importcredentials/cred management flow given the use of the installer pullsecret.

Reply:
ok sounds good 

I'll remove the /hold cancel once https://github.com/openshift/cluster-samples-operator/pull/104 merges so we get better operator e2e stability

Feel free to tag this gtm then at your discretion @bparees

Reply:
I picked https://github.com/openshift/cluster-samples-operator/pull/104 into here to get test runs restarted earlier